metagraph/src/graph/alignment/aligner_extender_methods.cpp

```diff
@@ -50,709 +55,695 @@ DefaultColumnExtender<NodeType>::DefaultColumnExtender(const DeBruijnGraph &grap
-                path.push_back(std::get<0>(best_node));
-                seq += std::get<1>(best_node);
-                assert((last_op == Cigar::MATCH)
-                    == (graph_.get_node_sequence(std::get<0>(best_node)).back()
-                        == extend_window_[pos - 1]));
-                switch (PS[pos - offset]) {
-                    case Extender::PREV: { best_node = prev; } break;
-                    case Extender::CUR: {} break;
-                    case Extender::NONE: { assert(false); }
-                }
-                --pos;
-                assert(pos);
-            } break;
-            case Cigar::INSERTION: {
-                assert(PS[pos - offset] == Extender::CUR);
-                while (last_op == Cigar::INSERTION) {
-                    last_op = OE[pos - offset];
-                    assert(last_op == Cigar::MATCH || last_op == Cigar::INSERTION);
-                    cigar.append(Cigar::INSERTION);
-                    --pos;
-                    assert(pos);
-                }
-            } break;
-            case Cigar::DELETION: {
-                while (last_op == Cigar::DELETION) {
-                    assert(table_.count(std::get<0>(best_node)));
-                    const auto &[S, E, F, OS, OE, OF, prev, PS, PF, offset, max_pos]
-                        = table_.find(std::get<0>(best_node))->second.first.at(std::get<2>(best_node));
-                    last_op = OF[pos - offset];
-                    assert(last_op == Cigar::MATCH || last_op == Cigar::DELETION);
-                    path.push_back(std::get<0>(best_node));
-                    seq += std::get<1>(best_node);
-                    cigar.append(Cigar::DELETION);
-                    switch (PF[pos - offset]) {
-                        case Extender::PREV: { best_node = prev; } break;
-                        case Extender::CUR: {} break;
-                        case Extender::NONE: { assert(false); }
-                    }
-                    prev_starts.emplace(best_node);
-                }
-            } break;
-            case Cigar::CLIPPED: { assert(false); }
-        }
+void DefaultColumnExtender
+::call_outgoing(node_index node,
+                size_t /* max_prefetch_distance */,
+                const std::function<void(node_index, char)> &callback) {
+    graph_->call_outgoing_kmers(node, [&](node_index next, char c) {
+        if (c != boss::BOSS::kSentinel)
+            callback(next, c);
+    });
+}
 
-        assert(pos);
-    }
+// allocate and initialize with padding to ensure that SIMD operations don't
+// read/write out of bounds
+template <class Column, typename... RestArgs>
+Column alloc_column(size_t size, RestArgs... args) {
+    Column column { {}, {}, {}, args... };
+    auto &[S, E, F, node, i_prev, c, offset, max_pos, trim] = column;
 
-    if (max_score < min_path_score)
-        return;
+    S.reserve(size + kPadding);
+    E.reserve(size + kPadding);
+    F.reserve(size + kPadding);
```

Comment:
Are you sure a multiplication factor wouldn't be better?
This additive will work well only if you indeed extend the vectors by a constant string not dependent on the initial size.

i.e., `2 * size`

Reply:
the SIMD code reads in 128-bit blocks, so I'm doing this padding to ensure that it doesn't read out of bounds.

To make things more compact, I could just as well just round the size up to the nearest multiple of 128 bits.

Reply:
why not resize then?

Reply:
also, write this in a comment

Reply:
I explained this in a comment, it's to allow for AlignedVector methods (like size() and push_back()) to still function properly.